gorm/transaction.go

```diff
@@ -41,9 +41,14 @@ func FromContext(ctx context.Context) (txn *Transaction, ok bool) {
 // It works as a singleton to prevent an application of creating more than one
 // transaction instance per incoming request.
 type Transaction struct {
-	mu      sync.Mutex
-	parent  *gorm.DB
-	current *gorm.DB
+	mu           sync.Mutex
+	parent       *gorm.DB
+	current      *gorm.DB
+	OnCommitHook func(ctx context.Context)
```

Comment:
let's make this unexportable (`onCommitHook`), and also make it a slice of `func(context.Context)` instead of a single hook. That way you can have multiple callbacks if needed

Reply:
let's also call it `afterCommit` instead of `onCommit` to make it clear when this fires